community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -129,44 +128,47 @@ public StateHandlingStatementOperations(
 
     // <Cursors>
     @Override
-    public NodeCursor nodeCursor( KernelStatement statement, long nodeId )
+    public Cursor<NodeItem> nodeCursor( KernelStatement statement, long nodeId )
     {
-        NodeCursor cursor = statement.getStoreStatement().acquireSingleNodeCursor( nodeId );
+        Cursor<NodeItem> cursor = statement.getStoreStatement().acquireSingleNodeCursor( nodeId );
         if ( statement.hasTxStateWithChanges() )
         {
-            return statement.txState().augmentSingleNodeCursor( cursor );
+            return statement.txState().augmentSingleNodeCursor( cursor, nodeId );
```

Comment:
While `acquireSingleNodeCursor` we already specified `nodeId` for which we would like to init cursor from store statement and we need to repeat that for `augmentSingleNodeCursor`. Is it ever possible that those id will be different?
If no, then can we init node cursor from TxState based on only cursor from store statement?